src/execution_plan/execution_plan.c

```diff
@@ -510,12 +510,25 @@ static void _ExecutionPlan_FreeInternals(ExecutionPlan *plan) {
 			QueryGraph_Free(plan->connected_components[i]);
 		}
 		array_free(plan->connected_components);
+		plan->connected_components = NULL;
 	}
 
-	QueryGraph_Free(plan->query_graph);
-	if(plan->record_map) raxFree(plan->record_map);
-	if(plan->record_pool) ObjectPool_Free(plan->record_pool);
-	if(plan->ast_segment) AST_Free(plan->ast_segment);
+	if(plan->query_graph) {
+		QueryGraph_Free(plan->query_graph);
+		plan->query_graph = NULL;
+	}
+	if(plan->record_map != NULL) {
+		raxFree(plan->record_map);
+		plan->record_map = NULL;
+	}
+	if(plan->record_pool != NULL) {
+		ObjectPool_Free(plan->record_pool);
+		plan->record_pool = NULL;
+	}
+	if(plan->ast_segment != NULL) {
+		AST_Free(plan->ast_segment);
+		plan->ast_segment = NULL;
+	}
```

Comment:
Why do we need to nullify these ?

Reply:
This was done for safety, since there were some problems with the freeing of the execution-plan.
This also exists in the refactoring of the execution-plan freeing mechanism.